toolchain/check/convert.cpp

```diff
@@ -494,12 +504,60 @@ static auto ConvertStructToStruct(Context& context, SemIR::StructType src_type,
     new_block.Set(i, init_id);
   }
 
-  return is_init ? context.AddNode(SemIR::StructInit{value.parse_node(),
+  CARBON_CHECK(!is_class || is_init)
+      << "Converting directly to a class value is not supported";
+  return is_class  ? context.AddNode(SemIR::ClassInit{value.parse_node(),
                                                      target.type_id, value_id,
                                                      new_block.id()})
-                 : context.AddNode(SemIR::StructValue{value.parse_node(),
-                                                      target.type_id, value_id,
-                                                      new_block.id()});
+         : is_init ? context.AddNode(SemIR::StructInit{value.parse_node(),
+                                                       target.type_id, value_id,
+                                                       new_block.id()})
+                   : context.AddNode(
+                         SemIR::StructValue{value.parse_node(), target.type_id,
+                                            value_id, new_block.id()});
+}
+
+// Performs a conversion from a struct to a struct type. Does not perform a
+// final conversion to the requested expression category.
+static auto ConvertStructToStruct(Context& context, SemIR::StructType src_type,
+                                  SemIR::StructType dest_type,
+                                  SemIR::NodeId value_id,
+                                  ConversionTarget target) -> SemIR::NodeId {
+  return ConvertStructToStructOrClass<SemIR::StructAccess>(
+      context, src_type, dest_type, value_id, target, /*is_class=*/false);
+}
+
+// Performs a conversion from a struct to a class type. Does not perform a
+// final conversion to the requested expression category.
```

Comment:
This seems to be an identical comment to `ConvertStructToStructOrClass`. The comments should probably call out a difference.

Reply:
Not identical -- that one says "to a struct or class type" and this one just says "to a class type". But I've changed the comment on the helper to say that it's a helper, which I think should resolve this.